src/sqlfluff/dialects/dialect_redshift.py

```diff
@@ -283,6 +283,81 @@ class CreateTableAsStatementSegment(BaseSegment):
     )
 
 
+@redshift_dialect.segment()
+class CreateExternalTableStatementSegment(BaseSegment):
+    """A `CREATE EXTERNAL TABLE` statement.
+
+    As specified in https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_EXTERNAL_TABLE.html
+    """
+
+    type = "create_external_table_statement"
+
+    match_grammar = Sequence(
+        "CREATE",
+        "EXTERNAL",
+        "TABLE",
+        Ref("TableReferenceSegment"),
+        Bracketed(
+            OneOf(
+                # Columns and comment syntax:
+                Delimited(
+                    Sequence(
+                        Ref("ColumnReferenceSegment"),
+                        Ref("DatatypeSegment"),
+                    ),
+                ),
+            )
+        ),
+        OneOf(Ref("PartitionedBySegment"), optional=True),
```

Comment:
```suggestion
        Ref("PartitionedBySegment", optional=True)
```
you can mark the ref itself as optional